overlord/hookstate/hookmgr.go

```diff
@@ -302,6 +302,8 @@ func (m *HookManager) runHook(task *state.Task, tomb *tomb.Tomb, snapst *snapsta
 		}
 	}
 
+	// XXX: why not return nil at this point if !(hookExists || mustHijack) ?
```

Comment:
This is a very good question, I'll dig a bit.

Reply:
Ok, I looked a bit on it and I *think* I *vaguely* (very vaguely) remember the idea behind this was to still allow our handler to do something even if there was no hook script. But I think that not being able to tell if a script was actually executed inside Done() handler is an oversight/limitation, so perhaps we should improve it.